### PR TITLE
feat: add a button to open dashboard group

### PR DESCRIPTION
**Pull request by feng-tao**
Signed-off-by: feng-tao <fengtao04@gmail.com>

### Summary of Changes

We redirect the group URL to search filter page in https://github.com/amundsen-io/amundsenfrontendlibrary/pull/655. Per UX design input, add a new button to open group url

### Tests

_What tests did you add or modify and why? If no tests were added or modified, explain why. Remove this line_

### Documentation

_What documentation did you add or modify and why? Add any relevant links then remove this line_

### CheckList

Make sure you have checked **all** steps below to ensure a timely review.

- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes, including screenshots of any UI changes.
- [ ] PR adds unit tests, updates existing unit tests, **OR** documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public python functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes all tests documented in the [developer guide](https://github.com/lyft/amundsenfrontendlibrary/blob/master/docs/developer_guide.md#testing)

### CODE REVIEW

File: amundsen_application/static/js/pages/DashboardPage/index.tsx [hunk 1]
```diff
@@ -205,6 +205,16 @@ export class DashboardPage extends React.Component<
           </div>
           {/* <div className="header-section header-links">links here</div> */}
           <div className="header-section header-buttons">
+            <a
+              id="dashboard-link"
```
Comment (Golodhros) [inline amundsen_application/static/js/pages/DashboardPage/index.tsx:209]: Could you use data-type instead of using the id?

Comment (feng-tao) [reply amundsen_application/static/js/pages/DashboardPage/index.tsx]: hey @Golodhros , pr is updated, could you share a bit more on when to use which one as I mostly follow the initial button?

Comment (Golodhros) [reply amundsen_application/static/js/pages/DashboardPage/index.tsx]: Thanks a lot Tao!

We should always use `data-` attributes to add information to the DOM. The previous implementation using an id wasn't the right way of doing things. From now on, we should always try to use data attributes. In modern frontend development, there is almost no reason to use an id attribute.

**Review by Golodhros — APPROVED** [review]
LGTM

File: amundsen_application/static/js/pages/DashboardPage/index.tsx (hunk 1 above)
Comment (feng-tao) [reply amundsen_application/static/js/pages/DashboardPage/index.tsx]: got it, thanks for the info!

**Comment by feng-tao** [comment]
thanks @Golodhros !

**Pull request merged**